### PR TITLE
fix: use tactic mctxBefore when extracting goalsBefore

### PR DESCRIPTION
**Pull request by adamtopaz**
## Summary
- fix the tactics extractor to evaluate `TacticInfo.goalsBefore` in `TacticInfo.mctxBefore`
- add a self-contained Lean regression fixture that reproduces the crash without mathlib
- add an integration regression test and wire it into `./run_tests`

## Details
The tactics extractor was rendering `goalsBefore` using the current `ContextInfo.mctx`.
For certain structural tactic nodes, especially nested `grind =>` sequences, Lean has already
removed the original goal metavariable from the current metavar context. That caused extraction to fail with:

```
unknown metavariable `?_uniq...`
```

This change mirrors Lean's own tactic formatting behavior by using `info.mctxBefore` when
interpreting `info.goalsBefore`.

The new regression fixture lives in:
- `LeanScoutTest/TacticsMetavarRegression.lean`

It is self-contained, uses only core Lean tactics, and reproduces the original failure mode.

## Testing
- `lake build`
- `./test/integration/test_tactics_metavar_regression.sh`
- `lake test`

### CODE REVIEW

File: LeanScout/DataExtractors/Tactics.lean [hunk 1]
```diff
@@ -27,21 +27,25 @@ public unsafe def tactics : DataExtractor where
...
-              usedConstants : $(consts.toList.map fun nm => s!"{nm}")
-            }
+        -- `goalsBefore` belongs to `info.mctxBefore`; using the current context's
+        -- metavar state can crash on structural nodes such as nested `grind` sequences.
+        let ctxBefore : Lean.Elab.ContextInfo := { ctxInfo with mctx := info.mctxBefore }
+        let goals : List Json ← ctxBefore.runMetaM' {} do
+          info.goalsBefore.mapM fun mvarId =>
+            mvarId.withContext do
+              let goal ← Lean.Meta.ppGoal mvarId
+              let t ← Lean.instantiateMVars <| .mvar mvarId
```
Comment (adamtopaz) [inline LeanScout/DataExtractors/Tactics.lean:68]: To do this we need to use the mctxAfter so that the mvar can be instantiated. Right now we use the mctxBefore, which means that the used constants won't be populated properly.

**Review by adamtopaz — COMMENTED** [review]
One possible issue noted.

File: LeanScout/DataExtractors/Tactics.lean (hunk 1 above)
Comment (adamtopaz) [reply LeanScout/DataExtractors/Tactics.lean:68]: Good catch. I verified that the previous revision emptied out `usedConstants` for the regression fixture because it evaluated both goal rendering and instantiation in `mctxBefore`. I updated the extractor to pretty-print `goalsBefore` in `mctxBefore`, but compute `usedConstants` by instantiating the same metavariable in `mctxAfter`. I also strengthened the regression test so it now checks that the `grindSeq` record has non-empty `usedConstants` and includes `False`. Re-ran `lake test` after the change.

**Comment by adamtopaz** [comment]
Follow-up: I hit another issue when running the tactics extractor over all of mathlib from `../scout_extraction`.

Root cause: in `--library` mode we were still treating dependency modules as raw `--read` files, which misses the per-module Lake `setup` data that `lake build` / `lean --setup` use. That caused some mathlib files (e.g. `Mathlib/RingTheory/TensorProduct/Maps.lean`) to fail even though the module itself builds fine.

I fixed this by:
- adding a `moduleData` library facet in `lakefile.lean` that emits each module's source path plus a materialized `setup.json`
- passing that setup file through library targets
- using Lean's setup-aware frontend path for input targets that have a setup file

Validation from `../scout_extraction` with 16-way parallelism:
- `lake run scout --command tactics --jsonl --parallel 16 --library Mathlib > /dev/null` ✅
- `lake run scout --command tactics --parquet --parallel 16 --dataDir /home/dev/scout_mathlib16_parquet --library Mathlib` ✅

The parquet run completed successfully and produced 128 parquet shards (~326M total).

**Comment by adamtopaz** [comment]
I also refactored the target model so setup-backed elaboration is represented explicitly instead of overloading `InputTarget`.

Changes:
- `InputTarget` is back to the plain pre-change shape (`path` only)
- added `SetupTarget` with `{ path, setupFile }`
- added `Target.setup`
- `--read` now always produces `.input`
- `--library` now always produces `.setup`
- setup-aware elaboration lives under `SetupTarget`, while `InputTarget` uses the original manual frontend path again

I added integration coverage to assert this wiring explicitly:
- `--read` logs an `input` target
- `--library` logs a `setup` target with a `setupFile`

Re-ran:
- `lake build` ✅
- `lake test` ✅
- `./test/integration/test_lean_orchestrator.sh` ✅
- `./test/integration/test_input_frontend_regression.sh` ✅
- `./test/integration/test_tactics_metavar_regression.sh` ✅

**Comment by adamtopaz** [comment]
I added source-location metadata to the `tactics` extractor to make spot checking much easier.

New fields on each tactic record:
- `module : String?`
- `startPos : { line : Nat, column : Nat }`
- `endPos : { line : Nat, column : Nat }`

Notes:
- line numbers are 1-based
- columns are 0-based
- `module` is populated for setup-backed/library extraction; for plain `--read` files without module setup it can be `null`

I also added coverage for this:
- Python extractor tests now check the new fields and exact spans for known tactics in `test_project`
- the metavariable regression test now checks source spans on the `have : n < x ∨ n < y` row
- README updated with the extended tactics schema

Validation:
- `lake build` ✅
- `uv run pytest test/extractors/test_tactics.py -v` ✅
- `./test/integration/test_tactics_metavar_regression.sh` ✅
- `lake test` ✅
